### DISCUSSION TITLE
Is there a way to make phased variant calls with samtools/bcftools?

### DISCUSSION THREAD

**Discussion by davidyuyuan** [samtools/bcftools#2149] [Q&A]
I built a tool chain to made variant calls against GRCh38. The VCFs are always unphased. I was unable to figure out how to make the variant phased with just samtools & bcftools. Is it feasible or I need to use other tools?

1. samtools phase to create phased bams
2. bcftools mpileup with 2 phased bams as input, then bcftools call --ploidy GRCh38
3. bcftools norm -m +any
4. bcftools filter...

What did I miss? Have I done something wrong?

**Comment by pd3** [comment]
Currently we have no tool to phase variants within bcftools and `samtools phase` is largely unsupported. Maybe you'd like to try WhatsHap 
https://www.biorxiv.org/content/10.1101/085050v2
https://github.com/whatshap/whatshap

**Reply by davidyuyuan** [reply]
Thanks, Petr. I was hoping not to drag in Whatshap or other tools. Also, I came across this review by PacBio comparing their own HiPhase with Whathap: https://github.com/PacificBiosciences/HiPhase/blob/main/docs/performance.md in April 2022. Both tools are updated regularly. I would need to compare them side-by-side myself and pick one then.
